cmd/kube-proxy/app/server.go

```diff
@@ -659,6 +659,8 @@ func getConntrackMax(config kubeproxyconfig.KubeProxyConntrackConfiguration) (in
 // CleanupAndExit remove iptables rules and exit if success return nil
 func (s *ProxyServer) CleanupAndExit() error {
 	encounteredError := userspace.CleanupLeftovers(s.IptInterface)
+	encounteredError = iptables.FlushIptablesChains(s.IptInterface) || encounteredError
+	encounteredError = ipvs.FlushIptablesChains(s.IptInterface) || encounteredError
```

Comment:
What's the reason for calling `FlushIptablesChains` separate from `CleanupLeftovers`? Would `iptables.CleanupLeftovers` error if we don't flush the IPVS chains first? IIRC they share the same chains.

Reply:
Because `iptables.CleanupLeftovers` want delete `KUBE-MARK-MASQ`, but `KUBE-LOAD-BALANCER` and `KUBE-NODE-PORT` link `KUBE-MARK-MASQ`, we shoud flush `KUBE-LOAD-BALANCER` and `KUBE-NODE-PORT` before delete `KUBE-MARK-MASQ`, `KUBE-LOAD-BALANCER` and `KUBE-NODE-PORT` is IPVS chains.